test/component2_suggestors/deprecated_lifecycle_suggestor_test.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react_codemod/src/component2_suggestors'
+    '/deprecated_lifecycle_suggestor.dart';
+import 'package:test/test.dart';
+
+import '../util.dart';
+import 'package:over_react_codemod/src/component2_suggestors/component2_constants.dart';
+
+main() {
+  group('ComponentWillMountMigrator', () {
+    final testSuggestor = getSuggestorTester(DeprecatedLifecycleSuggestor());
+
+    test('empty file', () {
+      testSuggestor(expectedPatchCount: 0, input: '');
+    });
+
+    test('no matches', () {
+      testSuggestor(
+        expectedPatchCount: 0,
+        input: '''
+          library foo;
+          var a = 'b';
+          class Foo {}
+        ''',
+      );
+    });
+
+    test('adds a FIXME comment for componentWillUpdate', () {
+      testSuggestor(
+        expectedPatchCount: 1,
+        input: '''
+          @Component2()
+          class FooComponent extends UiComponent2 {
+              componentWillUpdate(){}
```

Comment:
These methods - IRL - will nearly always have an `@override` annotation above them.  I think it would be a good idea to include that in these tests.  Or maybe tests with and without the annotation.

Reply:
Good call. I ensured there was basic coverage for both with and without the `@override`, but also added it to all other tests.